src/v2/guide/components.md

```diff
@@ -1,1481 +1,595 @@
 
-``` js
-Vue.component('button-counter', {
-  template: '<button v-on:click="incrementCounter">{{ counter }}</button>',
-  data: function () {
-    return {
-      counter: 0
-    }
-  },
-  methods: {
-    incrementCounter: function () {
-      this.counter += 1
-      this.$emit('increment')
-    }
-  },
-})
+## Dynamic Components
 
-new Vue({
-  el: '#counter-event-example',
-  data: {
-    total: 0
-  },
-  methods: {
-    incrementTotal: function () {
-      this.total += 1
-    }
-  }
-})
-```
+Sometimes, it's useful to dynamically switch between components, like in a tabbed interface:
 
 {% raw %}
-<div id="counter-event-example" class="demo">
-  <p>{{ total }}</p>
-  <button-counter v-on:increment="incrementTotal"></button-counter>
-  <button-counter v-on:increment="incrementTotal"></button-counter>
+<div id="dynamic-component-demo" class="demo">
+  <button
+    v-for="tab in tabs"
+    v-bind:key="tab"
+    class="dynamic-component-demo-tab-button"
+    v-bind:class="{ 'dynamic-component-demo-tab-button-active': tab === currentTab }"
+    v-on:click="currentTab = tab"
+  >
+    {{ tab }}
+  </button>
+  <component
+    v-bind:is="currentTabComponent"
+    class="dynamic-component-demo-tab"
+  ></component>
 </div>
 <script>
-Vue.component('button-counter', {
-  template: '<button v-on:click="incrementCounter">{{ counter }}</button>',
-  data: function () {
-    return {
-      counter: 0
-    }
-  },
-  methods: {
-    incrementCounter: function () {
-      this.counter += 1
-      this.$emit('increment')
-    }
-  }
-})
+Vue.component('tab-home', { template: '<div>Home component</div>' })
+Vue.component('tab-posts', { template: '<div>Posts component</div>' })
+Vue.component('tab-archive', { template: '<div>Archive component</div>' })
 new Vue({
-  el: '#counter-event-example',
+  el: '#dynamic-component-demo',
   data: {
-    total: 0
+    currentTab: 'Home',
+    tabs: ['Home', 'Posts', 'Archive']
   },
-  methods: {
-    incrementTotal: function () {
-      this.total += 1
+  computed: {
+    currentTabComponent: function () {
+      return 'tab-' + this.currentTab.toLowerCase()
     }
   }
 })
 </script>
+<style>
+.dynamic-component-demo-tab-button {
+  padding: 6px 10px;
+  border-top-left-radius: 3px;
+  border-top-right-radius: 3px;
+  border: 1px solid #ccc;
+  cursor: pointer;
+  background: #f0f0f0;
+  margin-bottom: -1px;
+  margin-right: -1px;
+}
+.dynamic-component-demo-tab-button:hover {
+  background: #e0e0e0;
+}
+.dynamic-component-demo-tab-button-active {
+  background: #e0e0e0;
+}
+.dynamic-component-demo-tab {
+  border: 1px solid #ccc;
+  padding: 10px;
+}
+</style>
 {% endraw %}
 
-In this example, it's important to note that the child component is still completely decoupled from what happens outside of it. All it does is report information about its own activity, just in case a parent component might care.
+The above is made possible by Vue's `<component>` element with the `is` special attribute:
```

Comment:
This seems to be a better way of organizing this, but we miss out on the good example that was provided before, where we show that `:is` can be really important if you need to preserve tags. We don't necessarily need to use the same example, but this becomes a bit of a shallow definition of what `is` does, IMO.

Reply:
I'm not sure I completely understand what you mean, but I'll be in a video call with you soon so can probably get clarification then. 🙂

Reply:
Sounds good, I'll find the line you deleted (which is in another file) that I'm referring to as well, so that what I'm referring to missing is more clear. This PR is a little funny to comment on because some of the changes are in two completely different places :)

Reply:
Ah, so if I'm understanding correctly, then this line:

> The custom component `<my-row>` will be hoisted out as invalid content, thus causing errors in the eventual rendered output. A workaround is to use the `is` special attribute:

Was not actually deleted but rewritten and moved around, so it's now:

> The custom component `<todo-item>` will be hoisted out as invalid content, causing errors in the eventual rendered output. Fortunately, the `is` special attribute offers a workaround:

It can definitely get confusing looking at the diff. 😅 I have that section a little lower, in the subsection about `DOM Template Parsing Caveats`. Do you feel like that's a good place for it? Is there any other important information you think I might have missed?

Reply:
Ah, ok there it is- so many moving parts, I missed where it went to. Thanks!